packages/nimble-components/src/table/types.ts

```diff
@@ -40,6 +40,18 @@ export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
 
+/**
+ * InternalTableRecord describes a hierarchical data structure that is used for
+ * the internal representation of the data, and allows us to represent data with
+ * parent-child relationships within Tanstack.
+ */
+export interface InternalTableRecord<
```

Comment:
Another possible name for this type would be `TableNode` or `TableRecordNode` to convey that it's a node in a tree. I think I like that slightly better because it conveys its purpose rather than its visibility (internal). But it's your call and I'm fine with the current name.

Reply:
Went with `TableNode` as that saved me four whole syllables!